include/setup.h

```diff
@@ -184,30 +184,40 @@ class Property {
 	const Changeable::Value change;
 };
 
-class Prop_int:public Property {
+class Prop_int : public Property {
 public:
-	Prop_int(std::string const& _propname,Changeable::Value when, int _value)
-		:Property(_propname,when) {
-		default_value = value = _value;
-		min = max = -1;
+	Prop_int(const std::string &name, Changeable::Value when, int val)
+	        : Property(name, when),
+	          min_value(-1),
+	          max_value(-1)
+	{
+		default_value = val;
+		value = val;
 	}
-	Prop_int(std::string const&  _propname,Changeable::Value when, int _min,int _max,int _value)
-		:Property(_propname,when) {
-		default_value = value = _value;
-		min = _min;
-		max = _max;
+
+	~Prop_int() override = default;
+
+	int GetMin() const { return min_value; }
+	int GetMax() const { return max_value; }
+
+	void SetMinMax(const Value &min, const Value &max)
+	{
+		min_value = min;
+		max_value = max;
 	}
-	int getMin() { return min;}
-	int getMax() { return max;}
-	void SetMinMax(Value const& _min,Value const& _max) {this->min = _min; this->max=_max;}
-	bool SetValue(std::string const& in);
-	~Prop_int(){ }
-	virtual bool CheckValue(Value const& in, bool warn);
-	// Override SetVal, so it takes min,max in account when there are no suggested values
-	virtual bool SetVal(Value const& in, bool forced,bool warn=true);
+
+	bool SetValue(const std::string &in) override;
+
+	bool CheckValue(const Value &in, bool warn) override;
+
+protected:
+	// Override SetVal, so it takes min,max in account when there are no
+	// suggested values
+	bool SetVal(const Value &in, bool forced, bool warn = true) override;
 
 private:
-	Value min,max;
+	Value min_value;
+	Value max_value;
```

Comment:
Do you know of any tools that can warning when naming violations pop up?

I would love a strong CI gate to keep the code moving in the right direction -- and would let us author PRs similar to the `effc++` cleanups. (I guess we don't need a tool for this.. but those warnings sure and handy and a gentle reminder to cleanup the code). 

The GUS code is actually packed with completely random conventions - so much so that I've struggled just to "match" it, because it's inconsistent even among itself :-)

Reply:
Well, pedantically speaking, as long as we don't have `using namespace std;` and names being aliased are not macros it's valid C/C++… but it's not helpful either way. I don't know any tools that prevent such kind of violations - I guess we'll need to look out for stuff like this during reviews.